backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/AstTreeParser.java

```diff
@@ -70,75 +83,162 @@ public List<EObject> parseJsonNode(final JsonNode astJson) {
-                            }
-                        }
                     }
                 } else {
-                    final Object target = eObject.eGet(reference, false);
-                    if (target instanceof InternalEObject eTarget && eTarget.eIsProxy()) {
-                        unresolvedReferences.add(new ProxiedReference(eObject, reference, eTarget));
-                    }
+                    this.messageReporter.error(MessageFormat.format("The transformer does not know how to handle key {0} with value {1} in {2}", key, value, this.logNameProvider.getName(eObject)));
                 }
             }
-        });
-        return unresolvedReferences;
+        }
     }
 
-    public void resolveAllReference(List<ProxiedReference> unresolvedReferences) {
-        List<ProxiedReference> unresolvedReferencesTmp = new ArrayList<>(unresolvedReferences);
+    public void resolveAllReference(List<ProxiedReference> unresolvedReferences0) {
+        // Sorts proxies to resolve by breadth-first search strategy. This way we resolve first the import
+        // generally located at top level part of the model. Those proxies are often used by lowest part of the
+        // model to resolve their link
+        List<ProxiedReference> unresolvedReferencesTmp = unresolvedReferences0.stream()
+                .sorted(new PoxiedReferenceComparator())
+                .toList();
+
         // There can be at first unsolvable references or aliases because they depend on another yet unresolved element.
         // Iterating over every unresolved reference until every single one is solved or no new resolution is possible.
+        int tryNb = 0;
         while (!unresolvedReferencesTmp.isEmpty()) {
+            tryNb++;
             List<ProxiedReference> proxiedReferences = unresolvedReferencesTmp.stream()
                     .filter(proxiedReference -> !this.proxyResolver.resolveProxy(proxiedReference))
                     .toList();
+            LOGGER.info(MessageFormat.format("{0} remaining proxy to resolve after {1} try", Integer.toString(proxiedReferences.size()), Integer.toString(tryNb)));
             if (proxiedReferences.equals(unresolvedReferencesTmp)) {
-                proxiedReferences.forEach(proxiedReference -> {
-                    LOGGER.error("Unable to find object with qualifiedName {} in namespace.", proxiedReference.targetProxy().eProxyURI().fragment());
-                });
+                this.printResolutionError(proxiedReferences);
                 break;
             } else {
                 unresolvedReferencesTmp = proxiedReferences;
             }
         }
     }
+
+    private void printResolutionError(List<ProxiedReference> proxiedReferences) {
+
+        for (ProxiedReference pr : proxiedReferences) {
+            String msg = MessageFormat.format("[Unable to resolve following proxy] Owner ''{0}'' to qualifiedName ''{1}'' for reference ''{2}''", this.logNameProvider.getName(pr.owner()),
```

Comment:
' is the escape character for MessageFormat so I need it if I want to add a ' in my final message.
Adding those ' in the final message ease the reading since qualified name can be very long, it helps knowing when it starts and when it ends.